src/Compilers/CSharp/Portable/Syntax/SyntaxFacts.cs

```diff
@@ -381,5 +381,65 @@ internal static bool IsVar(this Syntax.InternalSyntax.SyntaxToken node)
         {
             return node.Kind == SyntaxKind.IdentifierToken && node.ValueText == "var";
         }
+
+        /// <summary>
+        /// Figures out if this expression is a type in a deconstruction-declaration.
+        /// Outputs the VariableDeclarationSyntax directly containing it, if that is the case.
+        /// </summary>
+        internal static bool IsDeconstructionType(ExpressionSyntax expression, out SyntaxNode parent)
```

Comment:
I tried this again and I think it makes the code worse. Need to hold more locals, more casts, and also call site needs another dedicated local.
I'll leave as-is.